src/exp.jl

```diff
@@ -47,6 +47,55 @@ end
 
 Base.exp(A::Union{SkewHermitian,SkewHermTridiagonal}) = skewexp!(copyeigtype(A))
 
+function skewlog!(A::Union{SkewHermitian{T},SkewHermTridiagonal{T}}) where {T<:Real}
+    n = size(A, 1)
+    isodd(n) && throw("Logarithm of a singular matrix doesn't exist")
+    if typeof(A) <:SkewHermitian
+        vals, Qr, Qim = skeweigen!(A)
+    else
+        E = eigen!(A)
+        vals = E.values
+        Qr = real(E.vectors)
+        Qim = imag(E.vectors)
+    end
+    temp2 = similar(A, n, n)
+    Q1 = similar(A, n, n)
+    Q2 = similar(A, n, n)
+    r = similar(A, n)
+    θ = similar(A, n)
+
+    @simd for i = 1 : n
+        iszero(vals[i]) && throw("Logarithm of a singular matrix doesn't exist")
```

Comment:
You shouldn't call `throw` with a string — you should use an `Exception` type.   (You can throw a generic `ErrorException` with `error("foo")`.)

In this case, probably `throw(DomainError("logarithm o a singular matrix"))`.

(Although the standard matrix `log` just returns a matrix of `NaN` in this case.)